src/backends/interflop-vprec/interflop_vprec.c

```diff
@@ -1276,12 +1313,19 @@ static struct argp_option options[] = {
      "denormals-are-zero: sets denormals inputs to zero", 0},
     {key_ftz_str, KEY_FTZ, 0, 0, "flush-to-zero: sets denormal output to zero",
      0},
+    {key_preset_str, KEY_PRESET, "PRESET", 0,
+     "select a default PRESET setting among {binary16, binary32, binary64, "
+     "bfloat16, tensorfloat, fp24, PXR24}",
```

Comment:
Can you please document briefly in the help message for each preset, what are the range and precision.
For example, with a table

Format    (range, precision)
binary16 (5, 10)
binary32 (8, 23)
...

Reply:
The help message seems not appreciate table.

```c
    {key_preset_str, KEY_PRESET, "PRESET", 0,
     "select a default PRESET setting among {binary16, binary32, binary64, "
     "bfloat16, tensorfloat, fp24, PXR24}\n"
     "\tFormat (range, precision)\n"
     "\tbinary16    ( 5, 10)\n"
     "\tbinary32    ( 8, 23)\n"
     "\tbinary64    (11, 52)\n"
     "\tbfloat16    ( 8,  7)\n"
     "\ttensorfloat ( 8, 10)\n"
     "\tfp24        ( 7, 16)\n"
     "\tPXR24       ( 8, 15)",
     0}
   ```
   
   ```shell
         --preset=PRESET                                                                                                                            select a default PRESET setting among {binary16,
                             binary32, binary64, bfloat16, tensorfloat, fp24,
                             PXR24}
                                 Format (range, precision)
                                 binary16    ( 5, 10)
                                 binary32    ( 8, 23)
                                 binary64    (11, 52)
    bfloat16    ( 8,  7)
    tensorfloat ( 8, 10)
    fp24        ( 7, 16)
    PXR24       ( 8, 15)
 ```

Reply:
I did'nt think we can put a table in argp help message.

Reply:
Unfortunate :(

Yet it seems to work until line 5. Maybe group them in two columns ?
If still not working; you can format in line:

Format (range, precision) :  binary16 ( 5, 10)  binary32 (8, 23) binary64 (11, 52) ...

Reply:
I am forced to use the format in line. Because with two column it remains PXR24 which are not intended.

Reply:
Have you tried with these flags? https://www.gnu.org/software/libc/manual/html_node/Argp-User-Customization.html

Reply:
No, I didn't. But when I read it, it didn't seem like it would change my problem. Maybe I misunderstood some things. I will read it again and test it.

Reply:
By the way, fot the moment the help message is like that:
```
      --preset=PRESET        select a default PRESET setting among {binary16,
                             binary32, binary64, bfloat16, tensorfloat, fp24,
                             PXR24}
                             Format (range, precision) : binary16 (5, 10),
                             binary32 (8, 23), binary64 (11, 52), bfloat16 (8,
                             7), tensorfloat (8, 10), fp24 (7, 16), PXR24 (8,
                             15)
```

Reply:
I re-read ant I don't think these flags fix the problem.

I tried your flags but they didn't work. It changed nothing even with `rmargin=60` flags.

Reply:
It works on my side but it didn't solve the format issue. Actually these flags are passed by exporting the `ARG_HELP_FMT` environment variable. But I may have a solution for your problem.
You could add option with the `OPTION_DOC` for specifying that is only a documentation and not a real option. Code would be something like this:
```c
{key_preset_str, KEY_PRESET, "PRESET", 0, "select a default PRESET setting among:       format (range, precision)", 1},
{"Format (range, precision)", 0, "PRESET", OPTION_DOC, "", 1},
{"binary32  (8, 23)", 0, "PRESET", OPTION_DOC, "", 1},
```
It would give something like:
```bash
      --preset=PRESET        select a default PRESET setting among:
                             format (range, precision)
                             binary32 (8, 23)
```
Also setting the environement variable `setenv("ARGP_HELP_FMT", "doc-opt-col=29", 1);` before
parsing args to align the doc.


The only issue is that is that it makes code longer and DOC are sorted.

Reply:
It not really works for me.

With this code:
```c
static struct argp_option options[] = {                                                                                                                                                        
    /* --debug, sets the variable debug = true */                                                                                                                                              
    {key_prec_b32_str, KEY_PREC_B32, "PRECISION", 0,                                                                                                                                           
     "select precision for binary32 (PRECISION >= 0)", 0},                                                                                                                                     
    {key_prec_b64_str, KEY_PREC_B64, "PRECISION", 0,                                                                                                                                           
     "select precision for binary64 (PRECISION >= 0)", 0},                                                                                                                                     
    {key_range_b32_str, KEY_RANGE_B32, "RANGE", 0,                                                                                                                                             
     "select range for binary32 (0 < RANGE && RANGE <= 8)", 0},                                                                                                                                
    {key_range_b64_str, KEY_RANGE_B64, "RANGE", 0,                                                                                                                                             
     "select range for binary64 (0 < RANGE && RANGE <= 11)", 0},                                                                                                                               
    {key_input_file_str, KEY_INPUT_FILE, "INPUT", 0,                                                                                                                                           
     "input file with the precision configuration to use", 0},                                                                                                                                 
    {key_output_file_str, KEY_OUTPUT_FILE, "OUTPUT", 0,                                                                                                                                        
     "output file where the precision profile is written", 0},                                                                                                                                 
    {key_log_file_str, KEY_LOG_FILE, "LOG", 0,                                                                                                                                                 
     "log file where input/output informations are written", 0},                                                                                                                               
    {key_preset_str, KEY_PRESET, "PRESET", 0,                                                                                                                                                  
     "select a default PRESET setting among:", 1},                                                                                                                                             
    {"Format (range, precision)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                             
    {"binary16 (5, 10)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                      
    {"binary32 (8, 23)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                      
    {"binary64 (11, 52)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                     
    {"bfloat16 (8, 7)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                       
    {"tensorfloat (8, 10)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                   
    {"fp24 (7, 16)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                          
    {"PXR24 (8, 15)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                         
    {key_mode_str, KEY_MODE, "MODE", 0,                                                                                                                                                        
     "select VPREC mode among {ieee, full, ib, ob}", 0},                                                                                                                                       
    {key_err_mode_str, KEY_ERR_MODE, "ERROR_MODE", 0,                                                                                                                                          
     "select error mode among {rel, abs, all}", 0},                                                                                                                                            
    {key_err_exp_str, KEY_ERR_EXP, "MAX_ABS_ERROR_EXPONENT", 0,                                                                                                                                
     "select magnitude of the maximum absolute error", 0},                                                                                                                                     
    {key_instrument_str, KEY_INSTRUMENT, "INSTRUMENTATION", 0,                                                                                                                                 
     "select VPREC instrumentation mode among {arguments, operations, full}",                                                                                                                  
     0},                                                                                                                                                                                       
    {key_daz_str, KEY_DAZ, 0, 0,                                                                                                                                                               
     "denormals-are-zero: sets denormals inputs to zero", 0},                                                                                                                                  
    {key_ftz_str, KEY_FTZ, 0, 0, "flush-to-zero: sets denormal output to zero",                                                                                                                
     0},                                                                                                                                                                                       
    {0}};
```

I have this output where options are in mess:
```shell
Info [verificarlo]: loaded backend libinterflop_vprec.so
Usage: libinterflop_vprec.so [OPTION...] 

      --prec-input-file=INPUT   input file with the precision configuration to
                             use
      --prec-log-file=LOG    log file where input/output informations are
                             written
      --prec-output-file=OUTPUT   output file where the precision profile is
                             written
      --precision-binary32=PRECISION
                             select precision for binary32 (PRECISION >= 0)
      --precision-binary64=PRECISION
                             select precision for binary64 (PRECISION >= 0)
      --range-binary32=RANGE select range for binary32 (0 < RANGE && RANGE <=
                             8)
      --range-binary64=RANGE select range for binary64 (0 < RANGE && RANGE <=
                             11)
  -d, --daz                  denormals-are-zero: sets denormals inputs to zero
  -e, --error-mode=ERROR_MODE   select error mode among {rel, abs, all}
  -f, --ftz                  flush-to-zero: sets denormal output to zero
  -i, --instrument=INSTRUMENTATION
                             select VPREC instrumentation mode among
                             {arguments, operations, full}
  -m, --mode=MODE            select VPREC mode among {ieee, full, ib, ob}
      --max-abs-error-exponent=MAX_ABS_ERROR_EXPONENT
                             select magnitude of the maximum absolute error
      --preset=PRESET        select a default PRESET setting among:
                             bfloat16 (8, 7)
                             binary16 (5, 10)
                             binary32 (8, 23)
                             binary64 (11, 52)
                             Format (range, precision)
                             fp24 (7, 16)
                             PXR24 (8, 15)
                             tensorfloat (8, 10)
  -?, --help                 Give this help list
      --usage                Give a short usage message

Mandatory or optional arguments to long options are also mandatory or optional
for any corresponding short options.
```

And with this code:
```c
static struct argp_option options[] = {                                                                                                                                                        
    /* --debug, sets the variable debug = true */                                                                                                                                              
    {"Format (range, precision)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                             
    {"binary16 (5, 10)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                      
    {"binary32 (8, 23)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                      
    {"binary64 (11, 52)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                     
    {"bfloat16 (8, 7)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                       
    {"tensorfloat (8, 10)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                   
    {"fp24 (7, 16)", 0, "PRESET", OPTION_DOC, "", 1},                                                                                                                                          
    {"PXR24 (8, 15)", 0, "PRESET", OPTION_DOC, "", 1}, 
    {key_prec_b32_str, KEY_PREC_B32, "PRECISION", 0,                                                                                                                                           
     "select precision for binary32 (PRECISION >= 0)", 0},                                                                                                                                     
    {key_prec_b64_str, KEY_PREC_B64, "PRECISION", 0,                                                                                                                                           
     "select precision for binary64 (PRECISION >= 0)", 0},                                                                                                                                     
    {key_range_b32_str, KEY_RANGE_B32, "RANGE", 0,                                                                                                                                             
     "select range for binary32 (0 < RANGE && RANGE <= 8)", 0},                                                                                                                                
    {key_range_b64_str, KEY_RANGE_B64, "RANGE", 0,                                                                                                                                             
     "select range for binary64 (0 < RANGE && RANGE <= 11)", 0},                                                                                                                               
    {key_input_file_str, KEY_INPUT_FILE, "INPUT", 0,                                                                                                                                           
     "input file with the precision configuration to use", 0},                                                                                                                                 
    {key_output_file_str, KEY_OUTPUT_FILE, "OUTPUT", 0,                                                                                                                                        
     "output file where the precision profile is written", 0},                                                                                                                                 
    {key_log_file_str, KEY_LOG_FILE, "LOG", 0,                                                                                                                                                 
     "log file where input/output informations are written", 0},                                                                                                                               
    {key_preset_str, KEY_PRESET, "PRESET", 0,                                                                                                                                                  
     "select a default PRESET setting among:", 1},                                                                                                                                                                                                                                                                                     
    {key_mode_str, KEY_MODE, "MODE", 0,                                                                                                                                                        
     "select VPREC mode among {ieee, full, ib, ob}", 0},                                                                                                                                       
    {key_err_mode_str, KEY_ERR_MODE, "ERROR_MODE", 0,                                                                                                                                          
     "select error mode among {rel, abs, all}", 0},                                                                                                                                            
    {key_err_exp_str, KEY_ERR_EXP, "MAX_ABS_ERROR_EXPONENT", 0,                                                                                                                                
     "select magnitude of the maximum absolute error", 0},                                                                                                                                     
    {key_instrument_str, KEY_INSTRUMENT, "INSTRUMENTATION", 0,                                                                                                                                 
     "select VPREC instrumentation mode among {arguments, operations, full}",                                                                                                                  
     0},                                                                                                                                                                                       
    {key_daz_str, KEY_DAZ, 0, 0,                                                                                                                                                               
     "denormals-are-zero: sets denormals inputs to zero", 0},                                                                                                                                  
    {key_ftz_str, KEY_FTZ, 0, 0, "flush-to-zero: sets denormal output to zero",                                                                                                                
     0},                                                                                                                                                                                       
    {0}};
```

I have this output where `OPTION_DOC` are on the bottom of `--mode` options:
```shell
Info [verificarlo]: loaded backend libinterflop_vprec.so
Usage: libinterflop_vprec.so [OPTION...] 

  -d, --daz                  denormals-are-zero: sets denormals inputs to zero
  -e, --error-mode=ERROR_MODE   select error mode among {rel, abs, all}
  -f, --ftz                  flush-to-zero: sets denormal output to zero
  -i, --instrument=INSTRUMENTATION
                             select VPREC instrumentation mode among
                             {arguments, operations, full}
  -m, --mode=MODE            select VPREC mode among {ieee, full, ib, ob}
      --max-abs-error-exponent=MAX_ABS_ERROR_EXPONENT
                             select magnitude of the maximum absolute error
      --prec-input-file=INPUT   input file with the precision configuration to
                             use
      --prec-log-file=LOG    log file where input/output informations are
                             written
      --prec-output-file=OUTPUT   output file where the precision profile is
                             written
      --precision-binary32=PRECISION
                             select precision for binary32 (PRECISION >= 0)
      --precision-binary64=PRECISION
                             select precision for binary64 (PRECISION >= 0)
      --preset=PRESET        select a default PRESET setting among:
                             format (range, precision)
      --range-binary32=RANGE select range for binary32 (0 < RANGE && RANGE <=
                             8)
      --range-binary64=RANGE select range for binary64 (0 < RANGE && RANGE <=
                             11)
                             bfloat16 (8, 7)
                             binary16 (5, 10)
                             binary32 (8, 23)
                             binary64 (11, 52)
                             fp24 (7, 16)
                             PXR24 (8, 15)
                             tensorfloat (8, 10)
  -?, --help                 Give this help list
      --usage                Give a short usage message

Mandatory or optional arguments to long options are also mandatory or optional
for any corresponding short options.
```

It is complicated to have a table 😄

Reply:
It is because you need to set the same group for the doc and add them after the PRESET option. But it is a cosmetic thing, maybe fix this later.

Reply:
I think I do that with the first code of my comment ? (there are 2 different code)